runtime/src/main/scala/akka/grpc/internal/MetadataImpl.scala

```diff
@@ -159,12 +159,12 @@ class HeaderMetadataImpl(headers: immutable.Seq[HttpHeader] = immutable.Seq.empt
 
   override def getText(key: String): Option[String] =
     headers.reverseIterator.collectFirst {
-      case header if header.name == key => header.value
+      case header if header.name.equalsIgnoreCase(key) => header.value
```

Comment:
In akka-http, we often convert the `key` first to lower case and then compare to the stored lower-case name in the header (using `header.is()`), but maybe it doesn't matter for performance too much here?